lib/src/arithmetic/floating_point/floating_point_values/floating_point_value.dart

```diff
@@ -119,134 +118,185 @@ class FloatingPointValue implements Comparable<FloatingPointValue> {
+        _maxExp = computeMaxExponent(exponent.width) {
+    if (sign.width != 1) {
+      throw RohdHclException('FloatingPointValue: sign width must be 1');
+    }
+    if (constrainedMantissaWidth != null &&
+        mantissa.width != constrainedMantissaWidth) {
+      throw RohdHclException('FloatingPointValue: mantissa width must be '
+          '$constrainedMantissaWidth');
+    }
+    if (constrainedExponentWidth != null &&
+        exponent.width != constrainedExponentWidth) {
+      throw RohdHclException('FloatingPointValue: exponent width must be '
+          '$constrainedExponentWidth');
+    }
+  }
+
+  /// Constructs a [FloatingPointValue] with a sign, exponent, and mantissa
+  /// using one of the builders provided from [subtypeConstructorMap] if
+  /// available, otherwise using the default constructor.
+  factory FloatingPointValue.withMappedSubtype(
       {required LogicValue sign,
       required LogicValue exponent,
       required LogicValue mantissa}) {
-    if (exponent.width == FloatingPoint32Value.exponentWidth &&
-        mantissa.width == FloatingPoint32Value.mantissaWidth) {
-      return FloatingPoint32Value(
-          sign: sign, mantissa: mantissa, exponent: exponent);
-    } else if (exponent.width == FloatingPoint64Value._exponentWidth &&
-        mantissa.width == FloatingPoint64Value._mantissaWidth) {
-      return FloatingPoint64Value(
-          sign: sign, mantissa: mantissa, exponent: exponent);
-    } else {
-      return FloatingPointValue.withConstraints(
-          sign: sign, mantissa: mantissa, exponent: exponent);
-    }
-  }
+    final key = (exponentWidth: exponent.width, mantissaWidth: mantissa.width);
 
-  /// [FloatingPointValue] constructor from a binary string representation of
-  /// individual bitfields
-  factory FloatingPointValue.ofBinaryStrings(
-      String sign, String exponent, String mantissa) {
-    if (sign.length != 1) {
-      throw RohdHclException('Sign string must be of length 1');
+    if (subtypeConstructorMap.containsKey(key)) {
+      return subtypeConstructorMap[key]!(
+          sign: sign, exponent: exponent, mantissa: mantissa);
     }
 
     return FloatingPointValue(
-        sign: LogicValue.of(sign),
-        exponent: LogicValue.of(exponent),
-        mantissa: LogicValue.of(mantissa));
+        sign: sign, exponent: exponent, mantissa: mantissa);
   }
 
+  /// Converts this [FloatingPointValue] to a [FloatingPointValue] with the same
+  /// sign, exponent, and mantissa using the constructor provided in
+  /// [subtypeConstructorMap] if available, otherwise using the default
+  /// constructor.
+  FloatingPointValue toMappedSubtype() => FloatingPointValue.withMappedSubtype(
+      sign: sign, exponent: exponent, mantissa: mantissa);
+
+  /// [constrainedMantissaWidth] is the hard-coded mantissa width of the
+  /// sub-class of this floating-point value
+  @protected
+  int? get constrainedMantissaWidth => null;
```

Comment:
What is the purpose of these `constrained*Width` given that all subclasses hard code the widths?